provisioning/service/src/Models/TpmAttestation.cs

```diff
@@ -15,26 +15,6 @@ namespace Microsoft.Azure.Devices.Provisioning.Service
     /// </remarks>
     public class TpmAttestation : Attestation
     {
-        /// <summary>
-        /// Constructor for serialization and unit testing.
-        /// </summary>
-        /// <remarks>
-        /// This function will create a new instance of the TPM attestation
-        /// with both endorsement and storage root keys. Only the endorsement
-        /// key is mandatory.
-        /// </remarks>
-        /// <param name="endorsementKey">The string with the TPM endorsement key. It cannot be null or empty.</param>
-        /// <param name="storageRootKey">The string with the TPM storage root key. It can be null or empty.</param>
-        /// <exception cref="ArgumentNullException">If the provided <paramref name="endorsementKey"/> is null.</exception>
-        /// <exception cref="ArgumentException">If the provided <paramref name="endorsementKey"/> is empty or white space.</exception>
-        [JsonConstructor]
-        public TpmAttestation(string endorsementKey, string storageRootKey = null)
```

Comment:
Validations are not delegated entirely to service. We should be able to catch any leftover SDK validations with our tests.